signac_dashboard/modules/plotly_viewer.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2022 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import hashlib
+from typing import Callable, Dict, List, Tuple, Union
+
+import flask_login
+from flask import abort, jsonify, render_template, request
+from flask.views import View
+from jinja2.exceptions import TemplateNotFound
+from signac import Project
+from signac.contrib.job import Job
+
+from signac_dashboard.dashboard import Dashboard
+from signac_dashboard.module import Module
+
+
+class PlotlyView(View):
+    decorators = [flask_login.login_required]
+
+    def __init__(self, dashboard, args_function, context):
+        self.dashboard = dashboard
+        self.args_function = args_function
+        self.context = context
+
+    def dispatch_request(self):
+        if self.context == "JobContext":
```

Comment:
```suggestion
        if not flask_login.current_user.is_authenticated:
            return self.dashboard.login_manager.unauthorized(), 401
        if self.context == "JobContext":
```

Reply:
This is already covered by the `decorator` class attribute for the `PlotlyView` class [here](https://github.com/javierbg/signac-dashboard/blob/226111ef967cd294ebace776d82d0bca2b094d58/signac_dashboard/modules/plotly_viewer.py#L19).

It is documented [here](https://flask.palletsprojects.com/en/2.2.x/views/#view-decorators) and I have tested it myself by manually making the request without an authentication token.

Reply:
Thanks for the clarification. I missed that line.